lib/trie.js

```diff
@@ -0,0 +1,66 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function(path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+  if (parts.length) {
```

Comment:
Please use a reverse logic + early return to simplify the body of this function.

```js
if (!parts.length) {
  addVerbHandler(node, verb, handler);
  return this;
}

for (...) {
  ...
}
```